### ISSUE TITLE
review all of the fullvm flists

### ISSUE THREAD

**Issue by xmonader** [threefoldtech/tf-images#282]
[Zos](https://github.com/threefoldtech/zos/blob/main/pkg/primitives/vm/utils.go
) currently checks for the presence of specific files/symlinks to determine if a Flist represents a full VM. These files are:

- /image.raw (deprecated)
- /boot/vmlinuz
- /boot/initrd.img
However, the Ubuntu 24.04 full VM Flist [doesn't adhere]( https://hub.grid.tf/tf-official-vms/ubuntu-24.04-full.flist.md) to this specific file structure, resulting that zos treats it as a microvm and i'm expecting the rest to be the same

![image](https://github.com/user-attachments/assets/60a1c580-d602-4381-871c-2f694bcd1653)

**Comment by iwanbk** [comment]
and how the flist built?
If using some tool, might need to check whether there is bug in the tool

**Comment by PeterNashaat** [comment]
@iwanbk this is how fullvm is built https://github.com/threefoldtech/tf-images/tree/development/tfgrid3/ubuntu24.04/fullvm , https://github.com/threefoldtech/tf-images/tree/development/tfgrid3/ubuntu22.04/fullvm , And it's explained here in more details https://github.com/threefoldtech/zos/blob/main/docs/manual/zmachine/zmachine.md#vm

**Comment by PeterNashaat** [comment]
- @xmonader  Is the same issue in ubuntu 22 flist as well, as i can see it has /boot/vmlinuz
/boot/initrd.img files ?
<img width="914" alt="image" src="https://github.com/user-attachments/assets/f2b73848-3098-476d-a25d-eccd6172321e">

**Comment by PeterNashaat** [comment]
- Created a new ubuntu 24 flist  https://hub.grid.tf/petep.3bot/ubuntu-24.04_fullvm.flist
<img width="941" alt="image" src="https://github.com/user-attachments/assets/1add1a51-535f-49fb-a38d-8b1b44d220cc">
<img width="1051" alt="image" src="https://github.com/user-attachments/assets/f0cac629-5cc5-464e-b626-88bd2c157ff0">
- Changes pushed and created PR  /pull/284

**Comment by xmonader** [comment]
What is the status here? Have we reviewed the rest of the images?

**Comment by PeterNashaat** [comment]
- Fixed ubuntu 24 and verified ubuntu 22 is working fine

**Comment by xmonader** [comment]
moved this back in progress https://github.com/threefoldtech/tf-images/pull/309

**Comment by scottyeager** [comment]
I didn't see this before opening #308. I think we should consider standardizing all Ubuntu full VM images around using `linux-virtual` to install the kernel. As far as I can see it should be there for all of the versions we offer.

The approach of using `linux-modules-extra` to pull in the kernel itself technically worked in 22.04 (Jammy) at least, but it has the downside of increasing the image size substantially (~330mb) by installing the extra modules, which users can always install for themselves after deployment if needed. In current versions of 24.04 the modules remain compressed on disk, so the extra modules are only about 100mb but I still don't see a reason to ship these in the base image.

**Comment by PeterNashaat** [comment]
@scottyeager I completely agree—the linux-virtual kernel would indeed be the better fit. I previously tested the official Ubuntu 24 cloud image, which uses the linux-cloud kernel ([here](https://github.com/threefoldtech/tf-images/issues/228#issuecomment-2099972087)). But it required uisng [extract-vmlinux](https://github.com/threefoldtech/tf-images/issues/227#issuecomment-2075235236) and Zos didn't support kernel decompression. I'll proceed by creating a Ubuntu 22 Flist with the linux-virtual kernel and test it from there.

**Comment by PeterNashaat** [comment]
- Created a new ubuntu 22 flist https://hub.grid.tf/petep.3bot/ubuntu-22.04_fullvm.flist.md with linux-virtual kerenl 
- Tested new flist 
<img width="883" alt="image" src="https://github.com/user-attachments/assets/713defab-59f3-4a99-a478-f3ac56aa110d">
- /boot/ size is 0 !
<img width="1132" alt="image" src="https://github.com/user-attachments/assets/b7958cf9-d711-449d-8ecd-d22f851d791e">

@scottyeager i don't think it's working with ubuntu 22!

**Comment by xmonader** [comment]
@PeterNashaat so you moved it from verification to inprogress, what is going to happen with the ticket

**Comment by PeterNashaat** [comment]
yes i will be doing some tests on both ubuntu 24 flists to see which one is a better fit for us

**Comment by PeterNashaat** [comment]
- Testing flist https://hub.grid.tf/petep.3bot/ubuntu-24.04_fullvm.flist, built using linux-generic
- Then will Create and test flist with linux-virtual https://hub.grid.tf/petep.3bot/ubuntu-24.04_fv_linvirtual.flist

**Comment by PeterNashaat** [comment]
Created flist https://hub.grid.tf/petep.3bot/ubuntu-24.04_fv_linvirtual.flist and created a vm with it

**Comment by PeterNashaat** [comment]
#### A comparison on two kernels in Ubuntu 24: `linux-generic` and `linux-virtual`. and Logs are provided from the VMS:

- **Linux Generic Logs:** [Log Output for linux-generic kernel](https://gist.github.com/PeterNashaat/c3d064edb055e772e36c81331f9251fe)
- **Linux Virtual Logs:** [Log Output for linux-virtual kernel](https://gist.github.com/PeterNashaat/a2ae4af57dd3bf6417c0814ed41e4c69)

These flists were Used in a Full VMS on Mainnet

---

### **Examples of Extra Modules and Why They Are Not Needed**

The output from the `dpkg` and `find` commands ([linux extra modules](https://gist.github.com/PeterNashaat/f21eaf978f0d784228b4dd19fc136ace))

- **Accessibility Drivers:**
  - Examples: `speakup_acntsa.ko`, `speakup_soft.ko`
  - These are used for screen readers and are unnecessary 

- **Legacy Hardware Drivers:**
  - Examples: `pata_legacy.ko`, `pata_acpi.ko`
  - These drivers support older IDE (Parallel ATA) storage devices are unnecesesary

- **Specialized Hardware Accelerators:**
  - Examples: `habanalabs.ko`, `intel_vpu.ko`
  - Drivers for specific AI accelerators and VPUs are not required 

- **Size of Extra-modules**
  -  `107 MB` 
  
```bash
root@test24generic:~# dpkg-query -W --showformat='${Installed-Size}\n' linux-modules-extra-6.8.0-31-generic
109988
root@test24generic:~#
```
---

- Instead of using the full `linux-generic` package, I will use `linux-image-6.8.0-31-generic` for the flist. This ensures the inclusion of only the core kernel without the unnecessary extra modules.

If there are specific use cases requiring drivers from `linux-modules-extra`, those modules can be selectively added.

**Comment by PeterNashaat** [comment]
- Created a new vm with only linux-image-6.8.0-31-generic https://hub.grid.tf/petep.3bot/ubuntu-24.04_fullvm.flist
 Extra modules that are in linux-virtual and are not in linux-image-6.8.0-31-generic
```
xt_conntrack
nft_chain_nat
xt_MASQUERADE
nf_nat
nf_conntrack
nf_defrag_ipv6
nf_defrag_ipv4
bridge
stp
llc
xfrm_user
xfrm_algo
nft_compat
nf_tables
overlay
tls
```
- Linux-virtual has modules like `xt_conntrack, nf_nat, and nf_tables` are critical for advanced networking, such as setting up firewalls, NAT, and IPv6 handling.

**Comment by PeterNashaat** [comment]
- linux virtual kernel logs 
```
root@ubvirt24test2:~# uname -r
6.8.0-31-generic
root@ubvirt24test2:~#
root@ubvirt24test2:~# lsmod
Module                  Size  Used by
nls_iso8859_1          12288  0
kvm_intel             487424  0
kvm                  1437696  1 kvm_intel
irqbypass              12288  1 kvm
crct10dif_pclmul       12288  1
crc32_pclmul           12288  0
polyval_clmulni        12288  0
polyval_generic        12288  1 polyval_clmulni
ghash_clmulni_intel    16384  0
sha256_ssse3           32768  0
sha1_ssse3             32768  0
aesni_intel           356352  0
crypto_simd            16384  1 aesni_intel
cryptd                 28672  2 crypto_simd,ghash_clmulni_intel
binfmt_misc            24576  1
efi_pstore             12288  0
nfnetlink              20480  2
dmi_sysfs              24576  0
ip_tables              36864  0
x_tables               69632  1 ip_tables
autofs4                57344  2
virtiofs               36864  1
virtio_rng             12288  0
root@ubvirt24test2:~# dpkg -l | grep linux-image
ii  linux-image-6.8.0-31-generic   6.8.0-31.31                       amd64        Signed kernel image generic
ii  linux-image-virtual            6.8.0-31.31                       amd64        Virtual Linux kernel image
root@ubvirt24test2:~# dmesg | grep -i error
[    0.839735] RAS: Correctable Errors collector initialized.
root@ubvirt24test2:~#
```
- linux generic kernel logs 
```
root@ub24coretest2:~# uname -r
6.8.0-31-generic
root@ub24coretest2:~# lsmod
Module                  Size  Used by
nls_iso8859_1          12288  0
kvm_intel             487424  0
kvm                  1437696  1 kvm_intel
irqbypass              12288  1 kvm
crct10dif_pclmul       12288  1
crc32_pclmul           12288  0
polyval_clmulni        12288  0
polyval_generic        12288  1 polyval_clmulni
ghash_clmulni_intel    16384  0
sha256_ssse3           32768  0
sha1_ssse3             32768  0
aesni_intel           356352  0
crypto_simd            16384  1 aesni_intel
cryptd                 28672  2 crypto_simd,ghash_clmulni_intel
binfmt_misc            24576  1
efi_pstore             12288  0
nfnetlink              20480  2
dmi_sysfs              24576  0
ip_tables              36864  0
x_tables               69632  1 ip_tables
autofs4                57344  2
virtiofs               36864  1
virtio_rng             12288  0
root@ub24coretest2:~# dmesg | grep -i error
[    1.092603] RAS: Correctable Errors collector initialized.
root@ub24coretest2:~# dpkg -l | grep linux-image
ii  linux-image-6.8.0-31-generic   6.8.0-31.31                       amd64        Signed kernel image generic
root@ub24coretest2:~#
```
- Both `linux-image-generic` and `linux-virtual` use the same kernel version (6.8.0-31-generic) 
-  I recommend using `linux-image-generic` for the flist,  as it works fine wiht a wide range of user needs. While `linux-virtual` has a smaller footprint, it may lack drivers needed for specific use cases. Testing shows no significant errors in either configuration, but linux-image-generic ensures flexibility for users.

**Comment by PeterNashaat** [comment]
- Ubuntu 24 Flist : https://hub.grid.tf/petep.3bot/ubuntu-24.04_fullvm.flist
- Created PR https://github.com/threefoldtech/tf-images/pull/310/

**Comment by PeterNashaat** [comment]
@maxux could you please promote this flist https://hub.grid.tf/petep.3bot/ubuntu-24.04_fullvm.flist to ubuntu 24 official full vm

**Comment by maxux** [comment]
`petep.3bot/ubuntu-24.04_fullvm.flist ->
tf-official-vms/ubuntu-24.04-full.flist`

Promoted :)

**Comment by PeterNashaat** [comment]
- Getting errors in new flist 
```
/usr/sbin/grub-probe: error: failed to get canonical path of `vroot'.
dpkg: error processing package grub-pc (--configure):
 installed grub-pc package post-installation script subprocess returned error exit status 1
Errors were encountered while processing:
 grub-pc
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Comment by PeterNashaat** [comment]
- Fixed and verified https://hub.grid.tf/petep.3bot/ubuntu-24.04_fullvm.flist 
- @maxux could you please promote this flist to ubuntu 24 official full vm

<img width="1021" alt="Image" src="https://github.com/user-attachments/assets/4fe9d026-b403-4baf-9720-ae1d48dc1501" />

**Comment by maxux** [comment]
Promoted :)
https://hub.grid.tf/tf-official-vms/ubuntu-24.04-full.flist.md

**Comment by PeterNashaat** [comment]
Thanks
